### PR TITLE
Katib: Update Katib Config Guide with the new APIs 

### PR DESCRIPTION
**Pull request by andreyvelich**
Fixes: https://github.com/kubeflow/katib/issues/2186.

I updated guide with the new Katib Config APIs.

Please take a look.
/assign @tenzen-y @johnugeorge 

/cc @kubeflow/release-managers

### CODE REVIEW

**Comment by tenzen-y** [comment]
Ohhh, I would appreciate you for this effort. I really should have done this, but I could not prepare enough time 😓

**Comment by andreyvelich** [comment]
> Ohhh, I would appreciate you for this effort. I really should have done this, but I could not prepare enough time 😓

No worries, thanks again for the implementation! I think, we can merge this change before next release.

**Comment by tenzen-y** [comment]
> I think, we can merge this change before next release.

It makes sense.

File: content/en/docs/components/katib/katib-config.md [hunk 1]
```diff
@@ -6,436 +6,393 @@ weight = 70
...
+
+1. Katib controller: `controller`
+
+### Katib Certificate Generator Parameters
+
+The following parameters set in `.init.certGenerator` configure the Katib certificate generator:
+
+- `enable` - whether to enable Katib certificate generator.
+
+  The default value is `true`
```
Comment (tenzen-y) [inline content/en/docs/components/katib/katib-config.md:56]:
```suggestion
  The default value is `false`
```

This parameter's default is `false`: https://github.com/kubeflow/katib/blob/19268062f1b187dde48114628e527a2a35b01d64/pkg/apis/config/v1beta1/types.go#L88

File: content/en/docs/components/katib/katib-config.md [hunk 2]
```diff
@@ -6,436 +6,393 @@ weight = 70
...
+    limits:
+      cpu: -1
+      memory: -1
+      ephemeral-storage: -1
+  ```
+
+- `waitAllProcesses` - a flag to define whether the metrics collector should wait until all
+  processes in the Trial's training container are finished before start to collect metrics.
+
+  The default value is `true`
```
Comment (tenzen-y) [inline content/en/docs/components/katib/katib-config.md:197]: This parameter's default is nil, which means `false`.: 

- https://github.com/kubeflow/katib/blob/19268062f1b187dde48114628e527a2a35b01d64/pkg/apis/config/v1beta1/types.go#L125
- https://github.com/kubeflow/katib/blob/19268062f1b187dde48114628e527a2a35b01d64/pkg/apis/config/v1beta1/defaults.go#L167

**Review by tenzen-y — COMMENTED** [review]
Overall lgtm
Thanks!

File: content/en/docs/components/katib/katib-config.md (hunk 1 above)
Comment (andreyvelich) [reply content/en/docs/components/katib/katib-config.md:56]: Nice catch, Golang set `false` as default value for bool.

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/approve

/hold
for @johnugeorge

**Pull request merged**